### DISCUSSION TITLE
statusbarItemXYZ.title = something doesnt work as i think(?)

### DISCUSSION THREAD

**Discussion by GronbergInfo** [tui-cs/Terminal.Gui#2722] [General]
hi.

I have this menubar item called _statusPlaying_.
In the start of the program, i set _statusPlaying.title = ""_ and it becomes empty (as expected here)

Then when i click on the menu item or shortcut key for             _bitMenuItem.Action += new Action(StartMusic);_
It does start playing the wav i have in that function, however, now _statusPlaying.title = "something"_ does not update the statusbar item, UNTIL i click on the menubar.
Any idea what i am missing here as why it does not update the title?

**Comment by tznind** [comment]
Try calling `SetNeedsDisplay()` on the menu bar

**Comment by GronbergInfo** [comment]
> Try calling `SetNeedsDisplay()` on the menu bar

Can you be a little more specific? When should i run it? Is it enoug to run at the start/main once, or always when i need to update the text?

**Reply by GronbergInfo** [reply]
I gave it a try in the function i  have for the text update, and it works!
```
        private void UpdateMusic(string sText)
        {
            menuBar.SetNeedsDisplay();
            statusPlaying.Title = sText;
        }
```
